### PR TITLE
Add file persistence for infinispan

### PR DESCRIPTION
**Pull request by jbtrystram**
### Type of change

- Enhancement / new feature

### Description

Add a volumeClaimTemplate in infinispan deployment so when scaled each new replicas will have a volume attached to persist the data.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by kornys** [comment]
@jbtrystram @ctron does it really need 32Gi PV?

**Comment by lulf** [comment]
The pv size should probably be made configurable

**Comment by jbtrystram** [comment]
> The pv size should probably be made configurable

@lulf : isn't the yml file a config file already ? Iḿ happy to do it more configurable, what way do you have in mind?

File: templates/iot/examples/infinispan/manual/050-StatefulSet.yaml [hunk 1]
```diff
@@ -91,9 +91,23 @@ spec:
         volumeMounts:
         - mountPath: /opt/jboss/infinispan-server/standalone/configuration/custom
           name: infinispan-configuration
+#        - mountPath: /opt/jboss/infinispan-server/standalone/data/devices
+# this would be cleaner but causes infinispan to crash because it cannot write in
+#   /opt/jboss/infinispan-server/standalone/data
+        - mountPath: /data/
+          name: infinispan-data
```
Comment (jbtrystram) [inline templates/iot/examples/infinispan/manual/050-StatefulSet.yaml]: As described, mounting the volume in `/opt/jboss/infinispan-server/standalone/data/devices` prevent infinispan to write in the `data` folder for some reason... 
Any ideas @lulf @ctron  ?

**Comment by lulf** [comment]
@jbtrystram sorry, i somehow thought it was deployed by the iot operator and it was using a fixed size for it. If its just an example, then I think setting it to 32Gi is fine, but I think what @kornys is worried about is testing :)

File: templates/iot/examples/infinispan/manual/050-StatefulSet.yaml (hunk 1 above)
Comment (lulf) [reply templates/iot/examples/infinispan/manual/050-StatefulSet.yaml]: In what way is it prevented? Is it a permissions issue on the PV you are testing with?

**Comment by ctron** [comment]
I just tested it on my cluster, it works for me.

@jbtrystram Do you want to move it out of draft state?

**Comment by ctron** [comment]
@enmasse-ci run tests profile=iot

File: templates/iot/examples/infinispan/common/010-Config.yaml [hunk 2]
```diff
@@ -335,6 +335,15 @@ data:
                                    org.infinispan.query.indexmanager.InfinispanIndexManager
                                </property>
                            </indexing>
+                          <persistence passivation="false">
+                               <file-store
+                                        fetch-state="true"
+                                        read-only="false"
+                                        purge="false"
+                                        path="../../../../../../data/devices/">
```
Comment (dejanb) [inline templates/iot/examples/infinispan/common/010-Config.yaml]: Can we use absolute path here?

Comment (jbtrystram) [reply templates/iot/examples/infinispan/common/010-Config.yaml]: I tried that and could not get it to work .. It was always relative to `/opt/jboss/infinispan-server/standalone/data/`

Comment (jbtrystram) [reply templates/iot/examples/infinispan/common/010-Config.yaml]: @dejanb  i took another aproach with rocksDB and it is solved now :)

**Comment by dejanb** [comment]
@enmasse-ci run tests profile=iot

**Comment by jbtrystram** [comment]
It looks good on jenkins, why does it says failed here ?

**Pull request merged**